File: Baekjoon_Online_Judge/NO_10773/10773.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# ---------------------------- #
# | Created by happiness96   | #
# | Year 2020                | #
# | Month 08                 | #
# | Day 02                   | #
# | 10773 제로                | #
# ---------------------------- #


def run():
    K = int(r_input())

    stack = []

    for _ in range(K):
        number = int(r_input())

        if number:
            stack.append(number)
        
        else:
            stack.pop()
    
    print(sum(stack))


if __name__ == "__main__":
    run()

File: Baekjoon_Online_Judge/NO_10773/10773.cpp
#include <iostream>
#include <vector>
using namespace std;
/*
 ---------------------------- 
 | Created by happiness96   | 
 | Year 2020                | 
 | Month 08                 | 
 | Day 02                   | 
 | 10773 제로                | 
 ---------------------------- 
*/

int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    vector <int> v;

    int K;

    cin >> K;

    for(int i = 0; i < K; i ++){
        int num;

        cin >> num;
        
        if(num) v.push_back(num);
        else v.pop_back();
    }

    int total = 0;
    for(auto val : v) total += val;

    cout << total;

    return 0;
}
